x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportPutTrainedModelAction.java

```diff
@@ -186,11 +187,11 @@ protected void masterOperation(
                 return;
             }
 
-            Version minCompatibilityVersion = config.getModelDefinition().getTrainedModel().getMinimalCompatibilityVersion();
-            if (state.nodes().getMinNodeVersion().before(minCompatibilityVersion)) {
+            TransportVersion minCompatibilityVersion = config.getModelDefinition().getTrainedModel().getMinimalCompatibilityVersion();
+            if (state.getMinTransportVersion().before(minCompatibilityVersion)) {
                 listener.onFailure(
                     ExceptionsHelper.badRequestException(
-                        "Definition for [{}] requires that all nodes are at least version [{}]",
+                        "Cannot create model [{}] while cluster upgrade is in progress. All transport versions must be at least [{}]",
```

Comment:
These are user-facing error messages, and they won't have a clue what a transport version is, so we shouldn't mention it.

```suggestion
                        "Cannot create model [{}] while cluster upgrade is in progress.",
```